lms/js_config_types.py

```diff
@@ -0,0 +1,29 @@
+from typing import NotRequired, TypedDict
```

Comment:
Moved to it's own module here. THey way the current _js_config is setup, with all the code in __init__.py 
makes refactoring a bit tricky.